avm/res/web/site/main.bicep

```diff
@@ -294,6 +294,7 @@ module app_appsettings 'config--appsettings/main.bicep' = if (!empty(appSettings
     storageAccountUseIdentityAuthentication: storageAccountUseIdentityAuthentication
     appInsightResourceId: appInsightResourceId
     appSettingsKeyValuePairs: appSettingsKeyValuePairs
+    currentAppSettings: !empty(app.id) ? list('${app.id}/config/appsettings', '2023-12-01').properties : {}
```

Comment:
Should this be configurable? I fear that someone will open a ticket at some point to have full idempotency.

Reply:
@jikuja Certainly yes. IaC is about desired state and full update not incremental.
But I'm more concerned with a potential race condition.

Reply:
@jikuja Can you elaborate on what do you think when you say race condition ?

Reply:
> @jikuja Can you elaborate on what do you think when you say race condition ?

@pankajagrawal16 It seems the deployment is doing a fetch, then merge the properties. It is not an atomic operation. Two concurrent deployments (with different name) will fetch the same old values and only pushes its new properties. The last one to push will wins.

Reply:
@scrocquesel-ml150 Are you talking about multiple concurrent deployments of `avm/res/web/site/main.bicep` or concurrent deployment of `avm/res/web/site/main.bicep` main module and `avm/res/web/site/config--appsettings/main.bicep`?

Reply:
> @scrocquesel-ml150 Are you talking about multiple concurrent deployments of `avm/res/web/site/main.bicep` or concurrent deployment of `avm/res/web/site/main.bicep` main module and `avm/res/web/site/config--appsettings/main.bicep`?

two  `avm/res/web/site/main.bicep` will fetch the same existing values, then, after `resource app` is completed, each one will create a new `avm/res/web/site/config--appsettings/main.bicep` deployment with the same existing values and maybe different new values as `appSettingsKeyValuePairs`. I think if you want to be sure no race condition can occur, you can make an atomic deployment with the exact same deployment name because Azure will not permit to create a deployment of the same name if one exists and is not completed and will fail the deployment. So if you fetch the existing value and update the properties inside such a deployment, you have kind of a poor-man semaphore/lock.
I admit this is a race condition that is difficult to produce and everyone is using more or less this pattern without even knowing it exists. I guess it is a matter of complexity vs reliability.

Reply:
I don't see the issue you are describing here.

In case one is running the **same** deployments concurrently (it's already an anti pattern AFAIK) they will get the same `appSettingsKeyValuePairs` (i.e.: keeping values NOT defined in the bicep file, and merging them with the defined values from the bicep file) - so I don't see an issue or race condition: both deployment will result the same result een running concurrently.

In case somebody is tuning **different** deployments _concurrently_, then the last will win, aka the latest deployment will take precedence - which is normal. Also note, that was the behavior before this change already.